File: Hurb/Hurb/Model/Filters.swift
//
//  Filters.swift
//  Hurb
//
//  Created by Alexandre Papanis on 08/08/19.
//  Copyright © 2019 Papanis. All rights reserved.
//

import Foundation

struct Filters: Codable {
    let amenities: [FilterItem]?
    let attributes: [FilterItem]?
    let countries: [FilterItem]?
    let cities: [FilterItem]?
    let departureCities: [FilterItem]?
    let district: [FilterItem]?
    let neighborhood: [FilterItem]?
    let duration: [FilterItem]?
    let food: [FilterItem]?
    let people: [FilterItem]?
    let prices: [PriceItem]?
    let priceInterval: PriceInterval?
    let productType: [FilterItem]?
    let regulation: [FilterItem]?
    let rooms: [FilterItem]?
    let stars: [FilterItem]?
    let states: [FilterItem]?
}

struct FilterItem: Codable {
    let term: String?
    let filter: String?
    let count: Int?
}

struct PriceInterval: Codable {
    let min: Int?
    let max: Int?
    let filterPattern: String?
}

struct PriceItem: Codable {
    let min: Int?
    let maxExclusive: Int?
    let filter: String?
    let count: Int?
}

File: app/src/main/java/com/example/challenge_alpha/model/Filters.kt
package com.example.challenge_alpha.model

import com.google.gson.annotations.SerializedName

data class Filters(
    @SerializedName("amenities") val amenities: List<FilterDescription> = emptyList(),
    @SerializedName("attributes") val attributes: List<FilterDescription> = emptyList(),
    @SerializedName("countries") val countries: List<FilterDescription> = emptyList(),
    @SerializedName("cities") val cities: List<FilterDescription> = emptyList(),
    @SerializedName("departureCities") val departureCities: List<FilterDescription> = emptyList(),
    @SerializedName("duration") val duration: List<FilterDescription> = emptyList(),
    @SerializedName("food") val food: List<FilterDescription> = emptyList(),
    @SerializedName("people") val people: List<FilterDescription> = emptyList(),
    @SerializedName("prices") val prices: List<Prices> = emptyList(),
    @SerializedName("productType") val productType: List<FilterDescription> = emptyList(),
    @SerializedName("regulation") val regulation: List<FilterDescription> = emptyList(),
    @SerializedName("rooms") val rooms: List<FilterDescription> = emptyList(),
    @SerializedName("stars") val stars: List<FilterDescription> = emptyList(),
    @SerializedName("states") val states: List<FilterDescription> = emptyList()
    )

data class FilterDescription(
    @SerializedName("term") val description: String? = null,
    @SerializedName("filter") val filterString: String? = null,
    @SerializedName("count") val count: Int? = null
)

data class Prices(
    @SerializedName("min") val min: Int? = null,
    @SerializedName("maxExclusive") val maxExclusive: Int? = null,
    @SerializedName("filter") val filterString: String? = null,
    @SerializedName("count") val count: Int? = null
)